### ISSUE TITLE
Allow more flexible glob_patterns for images

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#301]
At the moment (and also after #300) we support globbing images like `glob("/some/path/*.png")`.
Can we also support more flexible patterns like `/some/path/*B03*.png`?

**Comment by tcompa** [comment]
As a self-reminder, a direct solution may look like the following.
In `create_ome_zarr.py`, we use `glob` three times, see slightly-modified code below:
```python
    # ...
    ext_glob_pattern = Path(input_paths[0]).name   # This could be e.g. "*.png"
    # ...
    for in_path in input_paths:
        input_filename_iter = in_path.parent.glob(ext_glob_pattern)
    # ...
    for plate in plates:
        # ...
        plate_image_iter = glob(f"{in_path}/{plate_prefix}_{ext_glob_pattern}")
        # ...
        for well in wells:
            well_image_iter = glob(f"{in_path}/{plate_prefix}_{well}{ext_glob_pattern}")
```
If we now set, for instance, `ext_glob_pattern="*B04*.png"`, the first glob will still work, but the other two will fail. The most direct solution is to have a double-pattern glob function, that produces two lists of filenames matching `pattern1= f"{in_path}/{plate_prefix}_{ext_glob_pattern}"` and `pattern2="f{str(in_path)/ext_glob_pattern}"`, and then takes their intersection.

A quick check can tell us the time it takes for this double-globbing on the largest dataset in https://github.com/fractal-analytics-platform/fractal/issues/213, but my first guess is that this would take a negligible amount of time.

Note that we are already using `glob.glob`, in the second and third calls (namely the ones that would be updated), meaning that we are already returning the full list (and not an generator, like `pathlib.Path.glob` does in the first call; we would need `glob.iglob` for that). This means that performance should not be any problem, because we would just call the same function twice and then call `set.intersection`.

**Comment by tcompa** [comment]
> A quick check can tell us the time it takes for this double-globbing on the largest dataset in [fractal-analytics-platform/fractal#213](https://github.com/fractal-analytics-platform/fractal/issues/213), but my first guess is that this would take a negligible amount of time.

TL;DR There are no performance issues for double-globbing, when looking at a folder with 163512 files.

---

Timings:
```
Found 163512 images matching "*.png", in 0.403 seconds
Examples:
    20200812-CardiomyocyteDifferentiation14-Cycle1_D05_T0001F070L01A02Z07C03.png
    20200812-CardiomyocyteDifferentiation14-Cycle1_E06_T0001F059L01A01Z28C01.png
    20200812-CardiomyocyteDifferentiation14-Cycle1_B05_T0001F051L01A02Z11C03.png

Found 6264 images matching "*_D09_*", in 0.249 seconds
Examples:
    20200812-CardiomyocyteDifferentiation14-Cycle1_D09_T0001F064L01A01Z20C02.png
    20200812-CardiomyocyteDifferentiation14-Cycle1_D09_T0001F062L01A01Z19C01.png
    20200812-CardiomyocyteDifferentiation14-Cycle1_D09_T0001F018L01A02Z24C03.png

Found 6264 images matching both "*.png" and "*_D09_*", in 0.690 seconds
Examples:
    20200812-CardiomyocyteDifferentiation14-Cycle1_D09_T0001F011L01A02Z13C03.png
    20200812-CardiomyocyteDifferentiation14-Cycle1_D09_T0001F005L01A02Z29C03.png
    20200812-CardiomyocyteDifferentiation14-Cycle1_D09_T0001F012L01A02Z13C03.png
```

Script
```python
import glob
import os
import time

folder = "/data/active/jluethi/20200810-CardiomyocyteDifferentiation14/Cycle1/images_renamed"
extension = "png"
pattern = "*_D09_*"

t0 = time.perf_counter()
all_images = glob.glob(f"{folder}/*.{extension}")
t1 = time.perf_counter()
print(f"Found {len(all_images)} images matching \"*.{extension}\", in {t1 - t0:.3f} seconds")
print("Examples:")
for ind in [0, 1, 2]:
    fname = os.path.basename(all_images[ind])
    print(f"    {fname}")
print()

t0 = time.perf_counter()
all_images_D09 = glob.glob(f"{folder}/{pattern}")
t1 = time.perf_counter()
print(f"Found {len(all_images_D09)} images matching \"{pattern}\", in {t1 - t0:.3f} seconds")
print("Examples:")
for ind in [0, 1, 2]:
    fname = os.path.basename(all_images_D09[ind])
    print(f"    {fname}")
print()

t0 = time.perf_counter()
all_images = glob.glob(f"{folder}/*.{extension}")
all_images_D09 = glob.glob(f"{folder}/{pattern}")
intersection = tuple(set(all_images).intersection(all_images_D09))
t1 = time.perf_counter()
print(f"Found {len(intersection)} images matching both \"*.{extension}\" and \"{pattern}\", in {t1 - t0:.3f} seconds")
print("Examples:")
for ind in [0, 1, 2]:
    fname = os.path.basename(intersection[ind])
    print(f"    {fname}")
print()
```

**Comment by tcompa** [comment]
The last comment somehow calls for a straightforward implementation with two arguments:
* image file extension (required, defaulting to tif)
* glob_pattern (optional, will be used for additional filtering)

Let's check it after #300

**Comment by jluethi** [comment]
As discussed at the call today, this double parameter, double glob approach sounds great! :)
